crates/starknet-devnet/tests/test_v3_transactions.rs

```diff
@@ -0,0 +1,472 @@
+                starknet_types::rpc::transaction_receipt::FeeUnit::FRI,
+                BlockTag::Latest,
+            )
+            .await
+            .unwrap();
+        assert!(estimate_fee.overall_fee > account_strk_balance);
+
+        match declaration.send().await.unwrap_err() {
+            starknet_rs_accounts::AccountError::Provider(ProviderError::StarknetError(
+                StarknetError::InsufficientAccountBalance,
+            )) => {}
+            other => panic!("Unexpected error: {:?}", other),
+        }
+    }
+
+    #[tokio::test]
+    async fn invoke_with_insufficient_gas_price_and_or_gas_units_should_fail() {
+        let devnet = BackgroundDevnet::spawn_with_additional_args(&[
+            "--gas-price-fri",
+            "10",
+            "--data-gas-price-fri",
+            "100",
+        ])
+        .await
+        .expect("Could not start Devnet");
+
+        let (signer, account_address) = devnet.get_first_predeployed_account().await;
+        let account: SingleOwnerAccount<
+            &starknet_rs_providers::JsonRpcClient<starknet_rs_providers::jsonrpc::HttpTransport>,
+            starknet_rs_signers::LocalWallet,
+        > = SingleOwnerAccount::new(
+            &devnet.json_rpc_client,
+            signer,
+            account_address,
+            constants::CHAIN_ID,
+            ExecutionEncoding::New,
+        );
+
+        transaction_with_less_gas_units_and_or_less_gas_price_should_return_error_or_be_accepted_as_reverted(
+            Action::Execution(vec![Call {
+                to: STRK_ERC20_CONTRACT_ADDRESS,
+                selector: get_selector_from_name("transfer").unwrap(),
+                calldata: vec![
+                    Felt::ONE,  // recipient
+                    Felt::ONE,  // low part of uint256
+                    Felt::ZERO, // high part of uint256
+                ],
+            }]),
+            Some(&account),
+            Option::<&OpenZeppelinAccountFactory<LocalWallet, JsonRpcClient<HttpTransport>>>::None,
+        )
+        .await;
+    }
+
+    #[tokio::test]
+    async fn deploy_account_with_insufficient_gas_price_and_or_gas_units_should_fail() {
+        let devnet = BackgroundDevnet::spawn_with_additional_args(&[
+            "--gas-price-fri",
+            "10",
+            "--data-gas-price-fri",
+            "100",
+        ])
+        .await
+        .expect("Could not start Devnet");
+
+        let signer = get_deployable_account_signer();
+        let factory = OpenZeppelinAccountFactory::new(
+            Felt::from_hex_unchecked(CAIRO_0_ACCOUNT_CONTRACT_HASH),
+            constants::CHAIN_ID,
+            signer,
+            devnet.clone_provider(),
+        )
+        .await
+        .unwrap();
+
+        let salt = Felt::THREE;
+        devnet
+            .mint_unit(
+                factory.deploy_v3(salt).address(),
+                1e18 as u128,
+                starknet_types::rpc::transaction_receipt::FeeUnit::FRI,
```

Comment:
Here and in other places, why not just import `starknet_types::rpc::transaction_receipt::FeeUnit`? That would really reduce the code size without sacrificing the clarity.

Reply:
[7d70b23](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/625/commits/7d70b2332b397ea50dd07d50430534100e447e01)